src/calculations.js

```diff
@@ -0,0 +1,521 @@
+  const totalSteps = steps * 1_000_000 / constraints
+  const fps = 25
+  const sec = totalSteps / fps
+  return Math.round(sec * 100) / 100
+}
+
+function _explosion(x, y, radius) {
+  let bombs = []
+  for (let i = 0; i < 100; i++) {
+    bombs.push({
+      x, y, i, radius
+    })
+  }
+  return bombs
+}
+
+function _addVectors(v1, v2) {
+  return [v1[0] + v2[0], v1[1] + v2[1]]
+}
+
+
+function _validateSeed(seed) {
+  const error = 'Seed must be a 32-byte value'
+  // ensure that the seed is a 32-byte value
+  if (typeof seed === 'string') {
+    if (seed.length !== 66) {
+      throw new Error(error + ' (1)')
+    }
+    // confirm that all characters are hex characters
+    if (seed.substring(2, 66).match(/[^0-9A-Fa-f]/)) {
+      throw new Error(error + ' (2)')
+    }
+    if (seed.substring(0, 2) !== '0x') {
+      throw new Error(error + ' (3)')
+    }
+    seed = BigInt(seed)
+  }
+  if (typeof seed === 'bigint') {
+    if (seed < 0n) {
+      throw new Error(error + ' (4)')
+    }
+    if (seed > 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFn) {
+      // if (seed > 115792089237316195423570985008687907853269984665640564039457584007913129639935n) {
+      throw new Error(error + ' (5)')
+    }
+  } else {
+    throw new Error(error + ' (6)')
+  }
+}
+
+
+// function checkCollision(body1, body2) {
+//   const distance = dist(body1.position.x, body1.position.y, body2.position.x, body2.position.y);
+//   const minDist = (body1.radius + body2.radius) / 4;
+
+//   if (distance < minDist) {
+//     // Calculate collision response
+//     const angle = atan2(body2.position.y - body1.position.y, body2.position.x - body1.position.x);
+//     const overlap = minDist - distance;
+
+//     const totalMass = body1.radius ** 2 + body2.radius ** 2;
+//     const overlapRatio1 = body2.radius / totalMass;
+//     const overlapRatio2 = body1.radius / totalMass;
+
+//     const deltaX = -cos(angle) * overlap;
+//     const deltaY = -sin(angle) * overlap;
+
+//     body1.position.x -= deltaX * overlapRatio1;
+//     body1.position.y -= deltaY * overlapRatio1;
+//     body2.position.x += deltaX * overlapRatio2;
+//     body2.position.y += deltaY * overlapRatio2;
+
+//     // Update velocities
+//     const angle1 = atan2(body1.velocity.y, body1.velocity.x);
+//     const angle2 = atan2(body2.velocity.y, body2.velocity.x);
+//     const speed1 = body1.velocity.mag();
+//     const speed2 = body2.velocity.mag();
+
+//     const newVelX1 = cos(angle1) * speed2;
+//     const newVelY1 = sin(angle1) * speed2;
+//     const newVelX2 = cos(angle2) * speed1;
+//     const newVelY2 = sin(angle2) * speed1;
+
+//     body1.velocity.set(newVelX1, newVelY1);
+//     body2.velocity.set(newVelX2, newVelY2);
+//   }
+// }
+
+
+
+
+export const utils = {
+  _smolr,
+  _convertBigIntToModP,
+  _approxDist,
+  _approxSqrt,
+  _approxDiv,
+  _calculateTime,
+  _explosion,
+  _addVectors,
+  _validateSeed,
+}
```

Comment:
I recommend exporting these functions directly so that they can be tree shaken, etc. e.g. this file would export `_smolr` etc rather than `utils.*`. Think it's the same amount or less code.